app/controllers/concerns/notifications_reader.rb

```diff
@@ -0,0 +1,13 @@
+module NotificationsReader
+  protected
+
+  # Mark each notifications associated with the item as read
+  def read_item_notifications(item)
+    item.comments.each do |comment|
+      comment.notifications.where(recipient: current_user, read_at: nil).each(&:read!)
```

Comment:
This generates one sql UPDATE call for every notification, which could potentially be a lot:

<img width="1204" alt="screen shot 2018-07-24 at 13 07 33" src="https://user-images.githubusercontent.com/2656485/43137107-be1cb324-8f42-11e8-94c4-26d82c577212.png">

You can reduce this to one SQL call by using `update_all`, but this has the disadvantage of not running any ActiveRecord validations/callbacks etc.

At the very least I think this `each` block should be wrapped in a transaction.

Reply:
Oh and `comment.notifications.where(recipient: current_user, read_at: nil)` can be shortened to `comment.notifications.unread.where(recipient: current_user)